### DISCUSSION TITLE
Removing eob signs in buffer

### DISCUSSION THREAD

**Discussion by asmodeus812** [nvim-tree/nvim-tree.lua#1582] [Q&A]
### Description

I would like to be able to clear the eob sign characters in the nvim tree buffer. Using auto command does not seem to work, and i have been facing this issue for quite some time without any progress. 

### Neovim version

```text
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by runner@fv-az164-457

Features: +acl +iconv +tui
```

### Operating system and version

wsl, ubuntu 20

### nvim-tree version

master

### Minimal config

```lua
None, using defaults, with empty setup call.
```

### Steps to reproduce

```
      augroup NVIMTREE
        autocmd!
        autocmd Filetype NvimTree setlocal fillchars=eob:\ ,fold:=
      augroup END
```

### Expected behavior

Eob characters should be cleared for the nvim tree buffer

### Actual behavior

Eob characters are still visible in the buffer.

**Comment by alex-courtis** [comment]
Autocommands are not particularly reliable due to inconsistent event ordering.

Try using the event API:
```lua
local Api = require('nvim-tree.api')
local Event = require('nvim-tree.api').events.Event

Api.events.subscribe(Event.TreeOpen, function()
	vim.opt_local.fillchars="eob: "
end)
```

**Comment by asmodeus812** [comment]
Seems to work fine, is that event called each time the tree is expanded/opened, is the buffer of the tree killed each time we toggle the tree off or is it just hidden, and the window pane is what gets destroyed ? 

 Another question is can that not be part of the other view properties on the tree config ? Like linenumber etc ?

**Comment by alex-courtis** [comment]
The buffer is destroyed whenever the tree is closed or its window is closed.

> Another question is can that not be part of the other view properties on the tree config ? Like linenumber etc ?

It's a very niche requirement, and the configuration would become very cluttered if we included every posible option.

The event exists for such niche use cases.
